frontend/public/components/monitoring/alerting.tsx

```diff
@@ -1301,15 +1301,24 @@ const AlertsPage_: React.FC<Alerts> = ({ data, loaded, loadError }) => (
 );
 const AlertsPage = withFallback(connect(alertsToProps)(AlertsPage_));
 
+const ruleHasAlertState = (rule: Rule, state: AlertStates): boolean =>
+  _.some(rule.alerts, { state });
+
+const ruleAlertStateFilter = (filter, rule: Rule) =>
+  _.some(rule.alerts, (a) => filter.selected.has(a.state)) || _.isEmpty(filter.selected);
+
 const rulesRowFilters: RowFilter[] = [
   {
-    filterGroupName: 'Rule State',
+    filter: ruleAlertStateFilter,
+    filterGroupName: 'Alert State',
+    isMatch: ruleHasAlertState,
```

Comment:
For this change, we can't use the `reducer` because we can't simply reduce each rule to a single value. Instead, we need to iterate through the rule's alerts looking for any one that matches the given alert state.